sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,472 @@
+"""
+This module can be used to solve 2D beam bending problems with
+singularity functions in mechanics.
+
+"""
+
+from __future__ import print_function, division
+
+from sympy.core import S, Symbol, diff
+from sympy.solvers import linsolve
+from sympy.printing import sstr
+from sympy.functions import SingularityFunction
+from sympy.core import sympify
+from sympy.integrals import integrate
+
+
+class Beam(object):
+    """
+    A Beam is a structural element that is capable of withstanding load
+    primarily by resisting against bending. Beams are characterized by
+    their cross sectional profile(Second moment of area), their length
+    and their material.
+
+
+    Examples
+    ========
+    There is a beam of length 4 meters. A constant distributed load of 6 Nm/m
+    is applied from half of the beam till the end. There are two supports below
+    the beam, one at the starting point and another at the ending point of the beam.
+    The deflection of the beam at the end is resticted.
+
+    >>> from sympy.physics.continuum_mechanics.beam import Beam
+    >>> from sympy import Symbol, Piecewise
+    >>> x = Symbol('x')
+    >>> E = Symbol('E')
+    >>> I = Symbol('I')
+    >>> b = Beam(4, E, I)
+    >>> b.apply_load(value=-9, start=4, order=-1)
+    >>> b.apply_load(value=-3, start=0, order=-1)
+    >>> b.apply_load(order=0, start=2, value=6)
+    >>> b.bc_deflection = [(4, 0)]
+    >>> b.boundary_conditions
+    {'deflection': [(4, 0)], 'moment': [], 'slope': []}
+    >>> b.load
+    -3*SingularityFunction(x, 0, -1) + 6*SingularityFunction(x, 2, 0) - 9*SingularityFunction(x, 4, -1)
+    >>> b.shear_force()
+    -3*SingularityFunction(x, 0, 0) + 6*SingularityFunction(x, 2, 1) - 9*SingularityFunction(x, 4, 0)
+    >>> b.bending_moment()
+    3*SingularityFunction(x, 0, 1) - 3*SingularityFunction(x, 2, 2) + 9*SingularityFunction(x, 4, 1)
+    >>> b.slope()
+    (3*SingularityFunction(x, 0, 2)/2 - SingularityFunction(x, 2, 3) + 9*SingularityFunction(x, 4, 2)/2 - 7)/(E*I)
+    >>> b.deflection()
+    (-7*x + SingularityFunction(x, 0, 3)/2 - SingularityFunction(x, 2, 4)/4 + 3*SingularityFunction(x, 4, 3)/2)/(E*I)
```

Comment:
I don't understand how this is more bad? Are you saying that you prefer the `__repr__` representation?